src/matter/cluster/server/OnOffServer.ts

```diff
@@ -0,0 +1,160 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    OnOffCluster, OnOffDelayedAllOffEffectVariant, OnOffDyingLightEffectVariant, OnOffEffectIdentifier
+} from "../OnOffCluster";
+import { ClusterServerHandlers } from "./ClusterServer";
+import { AttributeServer } from "./AttributeServer";
+
+/*
+TODO: Global Cluster fields needs to be added also here because, as discussed, based on the implementation.
+* Cluster Revision: 4 (If I get it right from the Specs - Application Cluster Specs 1.5.1)
+* FeatureMap: Bit 0 should be set when it is a "lighting" usecase AND Level Control cluster is supported
+* AttributeList:
+  * Supported always: onOff, onTime, offWaitTime
+  * Supported if we have a scene cluster: globalSceneControl
+  * Not supported: startUpOnOff
+* AcceptedCommandList:
+  * Supported always: Off, On, Toggle, OnWithTimedOff
+  * Supported if we have a scene cluster: onWithRecallGlobalScene
+  * Supported if we have a Level cluster: OffWithEffect
+* GeneratedCommandList: empty
+* EventList: empty
+* FabricIndex: empty
+ */
+
+/*
+TODO: If the Scenes server cluster is implemented on the same endpoint, the following extension field SHALL
+      be added to the Scene Table:
+      * OnOff
+ */
+
+export const OnOffClusterHandler: () => ClusterServerHandlers<typeof OnOffCluster> = () => ({
+    // TODO In fact for the Cluster server implementation we need "initialization code" or thinks like this.
+    //      I know that it is the wrong place in the "Handlers" object, but I don't know where to put it right now
+    //      because it should be "bound to the cluster instance" and not to the "cluster type".
+    //      Yes could also be build without the subscriptions, but so the code is separated from the other implementation
+    //      parts and if used as library and the "specific device implementation" manipulates onTime or offWaitTime then
+    //      it is all handled automatically and pot. not missed.
+    constructor: ({ attributes: { onOff, onTime, offWaitTime, globalSceneControl } }) => {
+        this.onTimeTimeout: NodeJS.Timeout | null = null;
+        onTime.addMatterListener((value) => {
+            if (this.onTimeTimeout) {
+                clearTimeout(this.onTimeTimeout);
+                this.onTimeTimeout = null;
+            }
+            if (value === 0) {
+                offWaitTime.set(0);
+                onOff.set(false);
+            }
+            if (value > 0 && onOff.get()) {
+                // TODO This is the 100% accurate impelementation right now ...
+                //      maybe could be optimized without a "100ms timeout loop",
+                //      but for now should do exactly what specs say
+                this.onTimeTimeout = setTimeout(() => {
```

Comment:
We need to do it that way if the onTime/offWaitTime values always need to be accurate when "get()" them. Alternative would be to be able to override the get implementation of an Attribute instance to calculate the current "rest time" value on get

Reply:
I am not sure why you need to add a listener on this attribute:

1. onTime.set(value): outside the OnWithTimedOff state sets the value but does nothing

2. onTime.set(value): during the OnWithTimedOff state modifies the value and this is used to know how long is left to wait before turning off (yes, with a 100ms accuracy, but I think this is fine given for the purpose of this cluster...)

For point 1, maybe there should be a wait to prevent set() to change the value outside a OnWithTimedOff cycle

Reply:
The reason I need it is because if on "on" command comes in then the offWaitTime is rest to 0 and such cases. So also other commands interact with onTime and offWaitTime.
So my idea was to have the logic in exactly one place - that means that the "attribute value handler" handles it based on the value which is set. So no other code need to care about the details.

The alternative would be to start and stop the Timer in several commands and this would be then spread everywhere (ok, mainly "stopping timer").